### ISSUE TITLE
sample_weight is accepted by fit() and silently ignored

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/jaxsr#19]
## Problem

`SymbolicRegressor.fit(X, y, sample_weight=...)` accepts the argument, documents it as "not yet implemented", and then **silently ignores it**. No warning, no error.

## Reproducer

Corrupt exactly the half of the data that is then down-weighted to near zero. A weighted fit should be unaffected; it is identical to the unweighted one.

```python
rng = np.random.default_rng(0)
X = rng.normal(size=(200, 1))
y = 3 * X[:, 0] + 0.1 * rng.normal(size=200)
y[:100] += 50                      # corrupt the first half
w = np.ones(200); w[:100] = 1e-6   # ...and down-weight exactly that half

lib = BasisLibrary(n_features=1, feature_names=['x']).add_constant().add_linear().add_polynomials(max_degree=3)
a = SymbolicRegressor(basis_library=lib, max_terms=2).fit(jnp.array(X), jnp.array(y))
b = SymbolicRegressor(basis_library=lib, max_terms=2).fit(jnp.array(X), jnp.array(y), sample_weight=jnp.array(w))
```

```
no weights  : y = 25.28 + 2.074*x^3
with weights: y = 25.28 + 2.074*x^3
identical -> sample_weight is silently ignored: True
```

Silently accepting an argument that changes nothing is worse than not accepting it: a user who passes measurement variances gets a plausible answer computed from an assumption they believe they overrode.

## Immediate, near-free fix

Raise `NotImplementedError`, or at minimum `warnings.warn`, until it is real. That alone would have saved a wrong analysis.

## Real implementation

Weighted least squares applied consistently, or it will be inconsistent in a way that is hard to see:

- coefficient fitting
- residual sums of squares and every derived metric
- information criteria — decide and document the effective-sample-size policy (`Σw` vs `n`), since it changes model selection
- selection strategies (greedy / exhaustive / LASSO screening)
- constraint refitting
- bootstrap and interval machinery

## Where it bites

Any problem with heteroscedastic measurements, replicate-derived variances, or regions of the design space that carry no information about the quantity of interest.

Concretely, in superposition discovery (#14) the identity `y_c = s'(c)·y_x` carries no information about `s'` wherever `y_x ≈ 0` — near the `G''` peak the log–log slope passes through zero, so those rows contribute noise and nothing else. The right treatment is a slope-aware weight; without working weights the prototype simply **drops** rows below a `|y_x|` threshold, which is blunt, discards partially-informative rows, and puts a hard threshold where a smooth taper belongs.

Note that duplicating rows is not an acceptable emulation — it corrupts the sample count that the information criteria depend on.
